src/test/scala/com/fortysevendeg/android/sqlite/SQLDroidDatabaseSpec.scala

```diff
@@ -0,0 +1,185 @@
+package com.fortysevendeg.android.sqlite
+
+import android.database.Cursor
+import android.database.sqlite.{SQLiteDatabaseLockedException, SQLiteDatabase}
+import org.specs2.mock.Mockito
+import org.specs2.mutable.Specification
+import org.specs2.specification.Scope
+
+trait SQLDroidDatabaseSpecification
+  extends Specification
+  with Mockito {
+
+  val databaseName = "database"
+
+  val timeout = 999
+
+  val retry = 500
+
+  val flags = 0
+
+  trait SQLDroidDatabaseScope
+    extends Scope {
+
+    val mockedDatabase = mock[SQLiteDatabase]
+
+    val cursor = mock[Cursor]
+
+    val sqlDroid = new SQLDroidDatabase(databaseName, timeout, retry, flags) {
+      override def openDatabase(name: String, flags: Int): SQLiteDatabase = mockedDatabase
+    }
+
+  }
+
+}
+
+class SQLDroidDatabaseSpec extends SQLDroidDatabaseSpecification {
+
+  "rawQuery" should {
+
+    "returns the Cursor when the method rawQuery in SQLiteDatabase works" in new SQLDroidDatabaseScope {
+      mockedDatabase.rawQuery(any, any) returns cursor
+      sqlDroid.rawQuery(sql = "", selectionArgs = Array()) shouldEqual cursor
+      there was one(mockedDatabase).rawQuery("", Array())
+    }
+
+    "returns the Cursor when the method rawQuery in SQLiteDatabase throws an exception in the first call and returns cursor in the second call" in new SQLDroidDatabaseScope {
+      mockedDatabase.rawQuery(any, any) throws new SQLiteDatabaseLockedException("Error") thenReturn cursor
+      sqlDroid.rawQuery(sql = "", selectionArgs = Array()) shouldEqual cursor
+      there was two(mockedDatabase).rawQuery("", Array())
+    }
+
+    "throws SQLiteDatabaseLockedException when rawQuery in SQLiteDatabase throws always an exception" in new SQLDroidDatabaseScope {
+      mockedDatabase.rawQuery(any, any) throws new SQLiteDatabaseLockedException("Error")
+      sqlDroid.rawQuery(sql = "", selectionArgs = Array()) must throwA[SQLiteDatabaseLockedException]
+      there was two(mockedDatabase).rawQuery("", Array())
+    }
+
+  }
+
+  "execSQL" should {
+
+    "executes successfully when the method execSQL in SQLiteDatabase works" in new SQLDroidDatabaseScope {
```

Comment:
Banner tip: `executes the SQL statement successful when the method execSQL in SQLite Database works fine`?